webapp/src/components/BuyEstatePage/BuyEstatePage.js

```diff
@@ -0,0 +1,248 @@
+  }
+
+  handleConfirm = () => {
+    const { publication, onConfirm } = this.props
+    onConfirm(publication)
+  }
+
+  renderLoading() {
+    return (
+      <div>
+        <Loader active size="massive" />
+      </div>
+    )
+  }
+
+  renderNotConnected() {
+    return (
+      <div>
+        <Container text textAlign="center" className="BuyEstatePage">
+          <Header as="h2" size="huge" className="title">
+            {t('asset_buy.buy_asset')}
+          </Header>
+          <p className="sign-in">
+            <T
+              id="global.sign_in_notice"
+              values={{
+                sign_in_link: (
+                  <Link to={locations.signIn()}>{t('global.sign_in')}</Link>
+                )
+              }}
+            />
+          </p>
+        </Container>
+      </div>
+    )
+  }
+
+  renderPage() {
+    const {
+      id,
+      wallet,
+      publication,
+      isDisabled,
+      isTxIdle,
+      onCancel
+    } = this.props
+    const { balance } = wallet
+
+    return (
+      <Estate id={id} ownerNotAllowed>
+        {estate => {
+          const allowance = this.getCurrentAllowance()
+
+          const price = parseFloat(publication.price)
+
+          const isNotEnoughMana = balance < price
+          const isNotEnoughAllowance = allowance < price
+          return (
+            <div className="BuyEstatePage">
+              {isNotEnoughMana || isNotEnoughAllowance
+                ? this.renderMessage()
+                : null}
+              <EstateModal
+                parcels={estate.data.parcels}
+                title={t('asset_buy.buy_asset', {
+                  asset_type: t('name.estate')
+                })}
+                subtitle={
+                  <T
+                    id="asset_buy.about_to_buy"
+                    values={{
+                      name: <EstateName estate={estate} />,
+                      price: publication ? (
+                        <React.Fragment>
+                          &nbsp;{t('global.for')}&nbsp;&nbsp;
+                          <span
+                            style={{
+                              display: 'inline-block',
+                              transform: 'translateY(3px)'
+                            }}
+                          >
+                            <Mana amount={publication.price} size={14} />
+                          </span>
+                        </React.Fragment>
+                      ) : (
+                        ''
+                      )
+                    }}
+                  />
+                }
+                onCancel={onCancel}
+                onConfirm={this.handleConfirm}
+                isDisabled={
+                  isDisabled || isNotEnoughMana || isNotEnoughAllowance
+                }
+                isTxIdle={isTxIdle}
+              />
+            </div>
+          )
+        }}
+      </Estate>
+    )
+  }
+
+  renderMessage() {
+    const { wallet, publication } = this.props
```

Comment:
The following +80 lines are pretty much the same as the ones in `BuyParcelPage`, could we componentize them?